lib/cocoapods/resolver.rb

```diff
@@ -380,6 +381,12 @@ def handle_resolver_error(error)
               'version requirement to your Podfile ' \
               "(e.g. `pod '#{name}', '#{lockfile_reqs.map(&:requirement).join("', '")}'`) " \
               "or revert to a stable version by running `pod update #{name}`."
+          elsif local_pod_parent
```

Comment:
I've tried to use `search_for(conflict.requirement)` to check this, but it returns no specification when conflict is caused by new dependency is not compatible with the locked dependency (e.g local pod has `'Expecta', '1.0.5'` and lockfile has `'Expecta', '1.0.3'`) because of the following line:

```
# Here it also adds the locking dependency dependency as a constraint
requirement = Requirement.new(dependency.requirement.as_list << requirement_for_locked_pod_named(dependency.name))
```

I can directly access to `find_cached_set` and check only for requested dependency's constraints, but it would be a somewhat duplicated code. Alternatively, I can add another `search_for` method taking 2 parameters, dependency and a boolean which will be used to decide if locked requirements should be taken into account. Both did not seem clean enough to me, but I think we should find a way to discriminate those cases and give the correct warning. I think this will come as an issue when somebody face with such a scenario (even if chance of facing with such a scenario is rare)